CMakeLists.txt

```diff
@@ -0,0 +1,203 @@
+  endif()
+  if(NOT Packet_ROOT)
+    set(Packet_ROOT ${PCAP_ROOT})
+  endif()
+endif()
+
+# Do we need Packet Packages on Windows ?
+if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+  find_package(Packet)
+  if(NOT Packet_FOUND)
+    message(FATAL_ERROR "Please specify Packet library -DPacket_ROOT=<NPCAP_SDK_PATH>")
+  endif()
+endif()
+
+find_package(PCAP)
+if(NOT PCAP_FOUND)
+  if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+    message(FATAL_ERROR "Please specify Npcap/WinPcap SDK directory with -DPCAP_ROOT=<PCAP_SDK_PATH>")
+  else()
+    message(FATAL_ERROR "PCAP library not found!")
+  endif()
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_IMMEDIATE_MODE "Enable PCAP immediate Mode (supported on libpcap>=1.5)" OFF)
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_IMMEDIATE_MODE)
+  if(NOT HAVE_PCAP_IMMEDIATE_MODE)
+    message(FATAL_ERROR "PCAP library doesn't have Immediate Mode support!")
+  endif()
+  add_definitions(-DHAS_PCAP_IMMEDIATE_MODE)
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_SET_DIRECTION
+       "Enable set direction for capturing incoming or outgoing packets (supported on libpcap>=0.9.1)" OFF)
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_SET_DIRECTION)
+  if(NOT HAVE_PCAP_DIRECTION)
+    message(FATAL_ERROR "PCAP library doesn't have Direction support!")
+  endif()
+  add_definitions(-DHAS_SET_DIRECTION_ENABLED)
+endif()
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+if(PCAPPP_TARGET_PLATFORM_WINDOWS AND PCAPPP_TARGET_COMPILER_MSVC)
+  # Other hacks for VStudio
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else()
+  if(NOT CMAKE_USE_PTHREADS_INIT)
+    message(FATAL_ERROR "Pthreads not found!")
+  endif()
+endif()
+
+if(PCAPPP_USE_DPDK)
+  find_package(DPDK)
+  if(NOT DPDK_FOUND)
+    message(FATAL_ERROR "Specify DPDK Library with -DDPDK_ROOT=<DPDK_PATH>")
+  endif()
+  add_definitions(-DUSE_DPDK)
+
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/mk/setup_dpdk.py ${CMAKE_CURRENT_BINARY_DIR} COPYONLY)
+endif()
+
+# Git Commit and Branch
+add_compile_definitions(GIT_COMMIT="${GIT_HASH}")
+add_compile_definitions(GIT_BRANCH="${GIT_BRANCH}")
+
+if(PCAPPP_USE_PF_RING)
+  find_package(PF_RING)
+  if(NOT PF_RING_FOUND)
+    message(FATAL_ERROR "Packet library -DPF_RING_ROOT=<PF_RING_PATH>")
+  endif()
+  add_definitions(-DUSE_PF_RING)
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release")
+endif()
+
+if(PCAPPP_TARGET_COMPILER_CLANG OR PCAPPP_TARGET_COMPILER_GCC)
+  add_compile_options(-Wall -g)
+endif()
+
+# Static linking GCC/Threads for minGW (Windows + GNU)
+if(PCAPPP_TARGET_PLATFORM_WINDOWS AND PCAPPP_TARGET_COMPILER_GCC)
+  add_link_options(-static-libgcc -static-libstdc++)
+  add_link_options(-static)
```

Comment:
The binary will use libgcc.dll and libstdc++.dll when running the final binary this libs are not automatically loaded by windows.
Or we will need to install those dll near the final executable

Reply:
Hm, this is just normal for dynamic linkage, isn't it?
And even Cmake has features for deploying runtime requirements.
How can I turn it off without patching?

Reply:
@dg0yt agree with you, I kept the same behavior as the legacy built system.
But we could of course deploy the runtime requirements instead of static linking